src/Plugins/ResizeMirror/ResizeMirror.js

```diff
@@ -0,0 +1,100 @@
+import AbstractPlugin from 'shared/AbstractPlugin';
+
+const onDragOver = Symbol('onDragOver');
+
+/**
+ * ResizeMirror default options
+ * @property {Object} defaultOptions
+ * @type {Object}
+ */
+export const defaultOptions = {};
+
+/**
+ * The ResizeMirror plugin resizes the mirror element to the dimensions of the draggable element that the mirror is hovering over
+ * @class ResizeMirror
+ * @module ResizeMirror
+ * @extends AbstractPlugin
+ */
+export default class ResizeMirror extends AbstractPlugin {
+  /**
+   * ResizeMirror constructor.
+   * @constructs ResizeMirror
+   * @param {Draggable} draggable - Draggable instance
+   */
+  constructor(draggable) {
+    super(draggable);
+
+    /**
+     * ResizeMirror options
+     * @property {Object} options
+     * @type {Object}
+     */
+    this.options = {
+      ...defaultOptions,
+      ...this.getOptions(),
+    };
+
+    this[onDragOver] = this[onDragOver].bind(this);
+  }
+
+  /**
+   * Attaches plugins event listeners
+   */
+  attach() {
+    this.draggable.on('drag:over', this[onDragOver]);
+    this.draggable.on('drag:over:container', this[onDragOver]);
+  }
+
+  /**
+   * Detaches plugins event listeners
+   */
+  detach() {
+    this.draggable.off('drag:over', this[onDragOver]);
+    this.draggable.off('drag:over:container', this[onDragOver]);
+  }
+
+  /**
+   * Returns options passed through draggable
+   * @return {Object}
+   */
+  getOptions() {
+    return this.draggable.options.resizeMirror || {};
+  }
+
+  /**
+   * Drag over handler
+   * @param {DragOverEvent} dragEvent
+   * @private
+   */
+  [onDragOver](dragEvent) {
```

Comment:
@tsov from here to the end of the file... I frigged with your code.

The issues I was experiencing before I made my changes:

1. Picking up a item would collapse its `width` and `height` _(it was unable to computer anything from what got picked up for `overElement`)_
2. Item would not get `width` + `height` until I moved `1px`.
3. I could not transition items, because the container appending was happening in the same frame as the `style` changes _(this is a no-no)_
4. Even with a single `animationFrame` separating the appending and the styling, that was not enough! I needed to double-wrap animation frames.

Reply:
Actually - the issue where the width / height isn't applied correctly on initial pickup is still present sometimes... dang!